### PR TITLE
FS/archive: wrap states into ArchiveManager class

### PR DESCRIPTION
**Pull request by wwylele**
Eliminates FS global state.

The ArchiveManager is placed in the top level class `Core::System` instead of FS because loader also uses it (note that his is currently done using `Core::System::GetInstance()`; will be changed after more refactor). This place can be temporary and might be moved to a more proper place in the following PRs.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4305)

### CODE REVIEW

File: src/core/hle/service/fs/archive.h [hunk 1]
```diff
@@ -49,258 +45,209 @@ enum class MediaType : u32 { NAND = 0, SDMC = 1, GameCard = 2 };
...
+     * @param id_code IdCode of the archive to open
+     * @param archive_path Path to the archive, used with Binary paths
+     * @return Handle to the opened archive
+     */
+    ResultVal<ArchiveHandle> OpenArchive(ArchiveIdCode id_code, FileSys::Path& archive_path);
+
+    /**
+     * Closes an archive
+     * @param handle Handle to the archive to close
+     */
```
Comment (B3n30) [inline src/core/hle/service/fs/archive.h]: Missing `@return ...`

File: src/core/hle/service/fs/archive.h [hunk 2]
```diff
@@ -49,258 +45,209 @@ enum class MediaType : u32 { NAND = 0, SDMC = 1, GameCard = 2 };
...
+     * @param low The low word of the SystemSaveData archive to delete
+     * @return ResultCode 0 on success or the corresponding code on error
+     */
+    ResultCode DeleteSystemSaveData(u32 high, u32 low);
+
+    /**
+     * Creates the SystemSaveData archive folder for the specified save data id
+     * @param high The high word of the SystemSaveData archive to create
+     * @param low The low word of the SystemSaveData archive to create
+     * @return ResultCode 0 on success or the corresponding code on error
```
Comment (B3n30) [inline src/core/hle/service/fs/archive.h]: Nit: The comments here about the ResultCode are very inconsistent

File: src/core/hle/service/fs/archive.h (hunk 1 above)
Comment (wwylele) [reply src/core/hle/service/fs/archive.h]: Was copy-pasting without touching 😂

File: src/core/hle/service/fs/archive.h (hunk 2 above)
Comment (wwylele) [reply src/core/hle/service/fs/archive.h]: ditto

**Comment by wwylele** [comment]
Sorry, not going to touch the existing doc in this PR because it is out of scope. Will do it in a "clean up all doc PR" when necessary.

**Pull request merged**